medicines/web/src/components/search-results/pagination.tsx

```diff
@@ -65,58 +65,54 @@ const StyledPagination = styled.nav`
   }
 `;
 
-const Pagination = (props: {
+interface IPaginationProps {
   searchTerm: string;
   resultCount: number;
   pageSize: number;
   currentPage: number;
   enabledDocTypes: DocType[];
-}) => {
+  handlePageChange: (num: number) => void;
+}
+
+const Pagination = (props: IPaginationProps) => {
   const pageCount = Math.floor(props.resultCount / props.pageSize) + 1;
   const { firstGroup, middleGroup, lastGroup } = getPaginationGroups(
     pageCount,
     props.currentPage,
   );
 
+  const getHandlePageChangeHandler = (num: number) => () =>
+    props.handlePageChange(num);
+
   const createPaginationButton = (page: number, i: number, array: number[]) => {
     const separator = i === array.length - 1 ? '' : <span>&ndash;</span>;
 
     if (page === props.currentPage) {
       return (
-        <li key={page}>
+        <li key={page + i}>
           {page}
           {separator}
         </li>
       );
     }
 
     return (
-      <li key={page}>
-        <Link href={getSearchPage(page)}>
-          <a>{page}</a>
-        </Link>
+      <li key={page + i}>
+        <a onClick={getHandlePageChangeHandler(page)}>{page}</a>
         {separator}
       </li>
     );
   };
 
-  const getSearchPage = (pageNo: number) => ({
-    pathname: '',
-    query: {
-      search: props.searchTerm,
-      page: pageNo,
-      doc: queryStringFromDocTypes(props.enabledDocTypes),
-    },
-  });
-
   return (
     <StyledPagination>
       <ul className="pagination">
         {props.currentPage !== 1 ? (
-          <li className="arrow">
-            <Link href={getSearchPage(props.currentPage - 1)}>
-              <a>Previous</a>
-            </Link>
+          <li
+            className="arrow"
+            onClick={getHandlePageChangeHandler(props.currentPage - 1)}
```

Comment:
I don't think `li[onClick] > a` is great web semantics. If we can't use `Link` as we were before, should we remove the `a` tag here?

Reply:
True - was just being lazy as the styling was all about the `a`. Have replaced this now and also added some nav related tags for better accessibility